internal/proxy/task_upsert.go

```diff
@@ -555,3 +603,79 @@ func (it *upsertTask) Execute(ctx context.Context) (err error) {
 func (it *upsertTask) PostExecute(ctx context.Context) error {
 	return nil
 }
+
+func (it *upsertTask) queryPK() (*milvuspb.QueryResults, error) {
+	log := log.Ctx(it.ctx)
+	log = log.With(
+		zap.Int64("collectionID", it.collectionID),
+		zap.String("collectionName", it.req.CollectionName),
+		zap.String("partitionName", it.req.PartitionName),
+	)
+	pkField, err := typeutil.GetPrimaryFieldSchema(it.schema)
+	if err != nil {
+		return nil, err
+	}
+	ids := it.result.IDs
+	expr := IDs2Expr(pkField.GetName(), ids)
+	partitionNames := make([]string, 0)
+	partitionNames = append(partitionNames, it.req.GetPartitionName())
+
+	// the GuaranteeTimestamp and TravelTimestamp can only set the channel have reached
+	// get the last timestamps of all pChannels
+	getAvalibleTickFunc := func(ticker channelsTimeTicker) (Timestamp, error) {
+		stats, ts, err := ticker.getMinTsStatistics()
```

Comment:
what does ticker.getMinTsStatistics mean?

I think the query should use the upsert ts?